src/main/java/securbank/dao/CreditCardDaoImpl.java

```diff
@@ -0,0 +1,56 @@
+package securbank.dao;
+
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import securbank.models.CreditCard;
+
+@Repository("creditCardDao")
+public class CreditCardDaoImpl extends BaseDaoImpl<CreditCard, UUID> implements CreditCardDao {
+	@Autowired
+	EntityManager entityManager;
+	
+	public CreditCardDaoImpl() {
+		super(CreditCard.class);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<CreditCard> findAll() {
+		return (List<CreditCard>) this.entityManager.createQuery("SELECT creditcard from CreditCard creditcard")
+				.getResultList();
+	
+	}
+
+	@Override
+	public CreditCard findByAccountNo(UUID accountID) {
+		try {
+			return (CreditCard) this.entityManager.createQuery("SELECT creditcard from CreditCard creditcard where (creditcard.accountID = :accountID) AND creditcard.active = 1")
+					.setParameter("accountID", accountID)
+					.getSingleResult();
+		}
+		catch(NoResultException e) {
+			// returns null if no user if found
+			return null;
+		}
+	}
+
+	@Override
+	public CreditCard findByCcId(UUID id) {
```

Comment:
@shivanijjwala This method is implemented in BaseDao. We don't need to redefine this